src/components/features/actionsMenu/MarkersTableActionsMenu.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import WarningActionDialog from '@/components/shared/WarningActionDialog'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { getWarehouseManagementApi } from '@/services/generated-api'
+
+import { SubmitHandler } from 'react-hook-form'
+import { NewMarkerFormData, newMarkerSchema } from '@/schemas/newMarkerSchema'
+import FormDialog from '@/components/shared/FormDialog'
+import NewMarkerForm from '@/components/features/forms/NewMarkerForm'
+import { BodyType } from '@/services/api'
+import { MarkerFormDto } from '@/services/model'
+import { useSnackbar } from '@/hooks/useSnackbar'
+import TableActionsMenu from './TableActionsMenu'
+
+interface MarkersTableActionsMenuProps {
+  id: number
+  name: string
+}
+
+export default function MarkersTableActionsMenu({ id, name }: MarkersTableActionsMenuProps) {
+  const { t: translate } = useTranslation()
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const [selectedOption, setSelectedOption] = React.useState<string | null>(null)
+  const { showSnackbar } = useSnackbar()
+
+  const handleClose = () => {
+    setSelectedOption(null)
+    setAnchorEl(null)
+  }
+
+  const onDiscardClick = () => {
+    handleClose()
+  }
+
+  const actionHandler = (option: string) => {
+    setSelectedOption(option)
+  }
+
+  const queryClient = useQueryClient()
+
+  const mutationDelete = useMutation({
+    mutationFn: (id: number) => getWarehouseManagementApi().deleteApiMarkerDeleteId(id),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['markers'] })
+      showSnackbar({
+        message: translate('newMarker.snackBar.messages.deleteMarker.success'),
+        type: 'success'
+      })
+    },
+    onError: () => {
+      showSnackbar({
+        message: translate('newMarker.snackBar.messages.deleteMarker.error'),
+        type: 'error'
+      })
+    }
+  })
+
+  const onConfirmClick = () => {
+    mutationDelete.mutate(id)
+    handleClose()
+  }
+
+  const mutationUpdate = useMutation({
+    mutationFn: ({ id, data }: { id: number; data: BodyType<MarkerFormDto> }) =>
+      getWarehouseManagementApi().putApiMarkerEditId(id, data),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['markers'] })
+      showSnackbar({
+        message: translate('newMarker.snackBar.messages.updateMarker.success'),
+        type: 'success'
+      })
```

Comment:
clear selected option on success and error on both update and delete. no need to keep these windows open.

Reply:
The dialog closes after 5 seconds. 
![image](https://github.com/InterCarsBulgaria-TechLeadAcademy/warehouse.management.fe/assets/123353314/ee05366f-5fcb-4bc7-95d1-e2ffb3c4fa2e)